### DISCUSSION TITLE
Not able to update slate value from useEffect

### DISCUSSION THREAD

**Discussion by smadapathi** [ianstormtaylor/slate#5109] [Q&A]
Hello Team,

Thanks for the library!
We are unable to update slate `value` from the useEffect hook. I have created a [sandbox link](https://codesandbox.io/s/little-wave-iu6w0z?file=/src/App.js) for reproducing the issue. If we comment line#24 in the above link, it works, but we were expecting it to work with setState in useEffect. 
We would like to update slate value asynchronously, any suggestions would be of a great help.

Thanks,

**Comment by SmilinBrian** [comment]
Unfortunately for a variety of reasons, you cannot just replace the `value` - it is only used as the initial value of the editor. Changing the children like in your example is not safe in all situations.

[This example](https://github.com/ianstormtaylor/slate/issues/4612#issuecomment-1041971128) shows one way to work around it by using on `key` to force the Editor to re-render when you need to change the editor contents 'externally'.
